test/cpp/api/operations.cpp

```diff
@@ -55,3 +55,33 @@ TEST_F(OperationTest, Cross) {
     ASSERT_TRUE(out.allclose(exp));
   }
 }
+
+TEST_F(OperationTest, Linear_out) {
+  {
+    const auto x = torch::arange(100., 118).resize_({3, 3, 2});
+    const auto w = torch::arange(200., 206).resize_({3, 2});
+    const auto b = torch::arange(300., 303);
+    auto y = torch::empty({3, 3, 3});
+    at::linear_out(y, x, w, b);
+    const auto y_exp = torch::tensor(
+        {{{40601, 41004, 41407}, {41403, 41814, 42225}, {42205, 42624, 43043}},
+         {{43007, 43434, 43861}, {43809, 44244, 44679}, {44611, 45054, 45497}},
+         {{45413, 45864, 46315}, {46215, 46674, 47133}, {47017, 47484, 47951}}},
+        torch::kFloat);
```

Comment:
nitpick: maybe we should test with one more dtype? This is where OpInfo would come pretty handy, IMO
What's the default type for c-tor ops such as `arange`, `empty`, btw? Wouldn't it be `kDouble`?

Reply:
I can do this in a separate PR.

Reply:
Will be handled by OpInfo PR